### PR TITLE
[core] Avoid extract from file footer when stats mode is none

### PR DESCRIPTION
**Pull request by Aitozi**
### Purpose

This could save one remote IO operation when stats mode is none

### CODE REVIEW

**Comment by wwj6591812** [comment]
+1

File: paimon-core/src/main/java/org/apache/paimon/io/StatsCollectingSingleFileWriter.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 public abstract class StatsCollectingSingleFileWriter<T, R> extends SingleFileWriter<T, R> {
 
     @Nullable private final SimpleStatsExtractor simpleStatsExtractor;
-    @Nullable private SimpleStatsCollector simpleStatsCollector = null;
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/io/StatsCollectingSingleFileWriter.java:46]: Can you keep it nullable, and just create a field `isStatsDisabled` from 
```
Arrays.stream(statsCollectors)
                        .allMatch(p -> p instanceof NoneSimpleColStatsCollector)
```

Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/io/StatsCollectingSingleFileWriter.java:46]: Fixed

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**